File: src/main/java/no/nav/syfo/config/ApplicationConfig.java
package no.nav.syfo.config;

import no.nav.syfo.exception.RestTemplateErrorHandler;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.web.client.RestTemplateBuilder;
import org.springframework.context.annotation.*;
import org.springframework.http.client.ClientHttpRequestInterceptor;
import org.springframework.web.client.RestTemplate;

import static java.util.Arrays.asList;


@Configuration
@Import({
        AdresseregisterConfig.class,
        FastlegeInformasjonConfig.class,
})
@Profile("remote")
public class ApplicationConfig {

    @Bean(name = "Oidc")
    public RestTemplate restTemplate(ClientHttpRequestInterceptor... interceptors) {
        RestTemplate template = new RestTemplate();
        template.setInterceptors(asList(interceptors));
        template.setErrorHandler(new RestTemplateErrorHandler());
        return template;
    }

    @Bean(name = "BasicAuth")
    public RestTemplate basicAuthRestTemplate(@Value("${srv.username}") String username,
                                              @Value("${srv.password}") String password) {
        return new RestTemplateBuilder()
                .basicAuthentication(username, password)
                .build();
    }

    @Bean(name = "default")
    public RestTemplate defaultRestTemplate() {
        return new RestTemplate();
    }
}

File: src/main/kotlin/no/nav/syfo/config/ApplicationConfig.kt
package no.nav.syfo.config

import no.nav.syfo.api.exception.RestTemplateErrorHandler
import org.springframework.context.annotation.*
import org.springframework.http.client.ClientHttpRequestInterceptor
import org.springframework.web.client.RestTemplate
import java.util.*

@Configuration
@Profile("remote")
class ApplicationConfig {
    @Bean(name = ["Oidc"])
    fun restTemplate(vararg interceptors: ClientHttpRequestInterceptor?): RestTemplate {
        val template = RestTemplate()
        template.interceptors = Arrays.asList(*interceptors)
        template.errorHandler = RestTemplateErrorHandler()
        return template
    }

   @Bean(name = ["default"])
    fun defaultRestTemplate(): RestTemplate {
        return RestTemplate()
    }
}
